### PR TITLE
Don't throw away error causes in TcpStreamLogic

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by jrudolph** [comment]
Turned out that the change to `StreamTcpException` slightly broke other things, so I simplified the change a bit and fixed the remaining broken tests.

File: akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala [hunk 1]
```diff
@@ -453,6 +451,20 @@ class TcpSpec extends StreamSpec("""
       }
     }
 
+    "provide full exceptions when connection attempt fails because name cannot be resolved" in {
+      val unknownHostName = "abcdefghijklmnopkuh"
+      import system.dispatcher
```
Comment (chbatey) [inline akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: is this used?

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala (hunk 1 above)
Comment (jrudolph) [reply akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala]: Nope, wasn't needed. Removed.

**Comment by patriknw** [comment]
I guess this is still ready to be merged. Running tests one more time.

**Comment by patriknw** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
Yep, if tests are still green it would be good to have.

**Pull request merged**